src/offset/library.cairo

```diff
@@ -629,6 +479,149 @@ namespace CarbonableOffseter {
         Claim.emit(address=caller, absorption=quantity, time=current_time);
         return ();
     }
+
+    func _deposit{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        from_token_id: Uint256, to_token_id: Uint256, to: felt, value: Uint256
+    ) -> (success: felt) {
+        alloc_locals;
+
+        // [Security] Start reetrancy guard
+        ReentrancyGuard.start();
+
+        // [Interaction] Transfer value from from_token_id to to_token_id
+        // let (old_deposited) = total_deposited();
+        let (carbonable_project_address) = CarbonableOffseter_carbonable_project_address_.read();
+        let (contract_address) = get_contract_address();
+        IERC3525.transferValueFrom(
+            contract_address=carbonable_project_address,
+            fromTokenId=from_token_id,
+            toTokenId=to_token_id,
+            to=to,
+            value=value,
+        );
+
+        // [Check] Transfer successful
+        // let (new_deposited) = total_deposited();
+        // let (expected_deposited) = SafeUint256.add(old_deposited, value);
+        // let (is_equal) = uint256_eq(new_deposited, expected_deposited);
+        // with_attr error_message("CarbonableOffseter: transfer failed") {
+        //     assert is_equal = TRUE;
+        // }
+
+        // [Effect] Store cumulated claimable
+        let (caller) = get_caller_address();
+        let (computed_claimable) = _claimable(address=caller);
+        let (stored_claimable) = CarbonableOffseter_claimable_.read(caller);
+        CarbonableOffseter_claimable_.write(caller, stored_claimable + computed_claimable);
+
+        // [Effect] Register the caller with the new value and the current timestamp
+        let (current_time) = get_block_timestamp();
+        let (previous_value) = CarbonableOffseter_registered_value_.read(caller);
+        let (new_value) = SafeUint256.add(previous_value, value);
+        CarbonableOffseter_registered_time_.write(caller, current_time);
+        CarbonableOffseter_registered_value_.write(caller, new_value);
+
+        // [Effect] Emit event
+        Deposit.emit(address=caller, value=value, time=current_time);
+
+        // [Effect] Register the caller if first deposit
+        let zero = Uint256(low=0, high=0);
+        let (is_zero) = uint256_eq(previous_value, zero);
+        if (is_zero == TRUE) {
+            let (index) = CarbonableOffseter_users_len_.read();
+            // Create new user
+            CarbonableOffseter_users_len_.write(index + 1);
+            CarbonableOffseter_users_.write(index, caller);
+            CarbonableOffseter_users_index_.write(caller, index);
+
+            // [Security] End reetrancy guard
+            ReentrancyGuard.end();
+            return (success=TRUE);
+        }
+
+        // [Security] End reetrancy guard
+        ReentrancyGuard.end();
+        return (success=TRUE);
```

Comment:
success can only be `TRUE`, right?

Reply:
Yes, actually the function fails or returns TRUE (Abdel legacy from project minter first implementation).